pkg/syncer/cnsoperator/controller/cnsregistervolume/util.go

```diff
@@ -95,25 +104,53 @@ func constructCreateSpecForInstance(r *ReconcileCnsRegisterVolume, instance *cns
 }
 
 // getK8sStorageClassName gets the storage class name in K8S mapping the vsphere
-// storagepolicy id.
-func getK8sStorageClassName(ctx context.Context, k8sClient clientset.Interface, storagePolicyID string) (string, error) {
+// storagepolicy id. The policy must also be assigned to the passed namespace.
+func getK8sStorageClassName(ctx context.Context, k8sClient clientset.Interface, storagePolicyID string, namespace string) (string, error) {
 	log := logger.GetLogger(ctx)
 	scList, err := k8sClient.StorageV1().StorageClasses().List(ctx, metav1.ListOptions{})
 	if err != nil {
 		msg := fmt.Sprintf("Failed to get Storageclasses from API server. Error: %+v", err)
 		log.Error(msg)
 		return "", errors.New(msg)
 	}
+	var scName string
 	for _, sc := range scList.Items {
 		scParams := sc.Parameters
 		for paramName, val := range scParams {
 			param := strings.ToLower(paramName)
 			if param == common.AttributeStoragePolicyID && val == storagePolicyID {
-				return sc.Name, nil
+				scName = sc.Name
+				break //There will be only one storage class in the cluster with a given policy ID.
 			}
 		}
 	}
-	msg := fmt.Sprintf("Failed to find K8s Storageclass mapping storagepolicyId: %s", storagePolicyID)
+
+	/*
+		Resource Quotas
+			Name:                                                                   <namespace>-storagequota
+			Resource                                                                Used  Hard
+			--------                                                                ---   ---
+			<storage-class-name>.storageclass.storage.k8s.io/requests.storage  		0     5Gi
+	*/
+	quota, err := k8sClient.CoreV1().ResourceQuotas(namespace).Get(ctx, namespace+storageRQNameSuffix, metav1.GetOptions{})
```

Comment:
There could be multiple `ResourceQuota` objects in a namespace. Should we not list them all?

Reply:
But we are only interested in the storage resource quota which is named `<namespace>-storagequota`. So fetching the specific object here. Do you think storage class resources can be listed under any other `ResourceQuota` object?

Reply:
Kubernetes allows creating multiple `ResourceQuota`. I think its better we list all `ResourceQuota`, go through each and look for whether the storage class is used or not. I think currently WCP is setting a single ResourceQuota, but nothing it stopping WCP or any other service to create more `ResourceQuota` objects. 

We can commit the code you have too with the caveat that we will have to revisit if the current assumptions you are making is incorrect.

Reply:
Makes sense. I have made the changes to go through all resource quotas.